package.json

```diff
@@ -163,6 +163,7 @@
     "semver": "7.5.4",
     "shelljs": "^0.8.5",
     "single-line-log": "1.1.2",
+    "slugify": "^1.6.6",
```

Comment:
Do we _have_ to introduce this?

Reply:
We don't _have_ to but it really does cover all the bases for slugification. I.e..: it has character replacement lists for 

Technically we can code our own slugify algorithm

Reply:
> Technically we can code our own slugify algorithm

would definitely reduce cognitive load

Reply:
> We are tempted to not reinvent the wheel so strong that we are ready to import large, heavy libraries to use a small function that we could easily write by ourselves. It introduces unnecessary dependencies and bloated code. Make informed decisions about when to import external libraries and when it is more appropriate to write concise, self-contained code snippets to accomplish smaller tasks.

https://github.com/zakirullin/cognitive-load#abusing-dry-principle

Reply:
I understand this reasoning but the reason I decided to use this dependency anyway is because it's [not a large or bloated library](https://github.com/simov/slugify/blob/master/slugify.js). Of course, we can be super specific in this case and make a simpler(smaller) logic specifically to slugify domain names but in that case, it can't be used elsewhere

Reply:
> I understand this reasoning but the reason I decided to use this dependency anyway is because it's not a large or bloated library.

I think one of the dangers of still using this is that libraries do tend to get bloated or [riskier](https://www.bleepingcomputer.com/news/security/dev-corrupts-npm-libs-colors-and-faker-breaking-thousands-of-apps/) in the future, so it becomes more about future proofing in that case. I just feel like this is a small enough task to code in ourselves - but feel free to ignore if its going to be heavy lift.

Reply:
> We are tempted to not reinvent the wheel so strong that we are ready to import large, heavy libraries to use a small function that we could easily write by ourselves. It introduces unnecessary dependencies and bloated code. **Make informed decisions about when to import external libraries** and when it is more appropriate to write concise, self-contained code snippets to accomplish smaller tasks.

I bolded the part where I think we should consider or not. Is importing this library an informed decision? Is not importing the library also an informed decision?

> [riskier](https://www.bleepingcomputer.com/news/security/dev-corrupts-npm-libs-colors-and-faker-breaking-thousands-of-apps/) in the future,

The problem linked is easily mitigated by having a lockfile i.e. `yarn.lock, npm-shrinkwrap.json, package-lock.json` - each package version includes an integrity hash to prevent hackers from swapping the package with something else.

> I just feel like this is a small enough task to code in ourselves

There seems to be a lot of [edge cases](https://github.com/simov/slugify/blob/master/test/slugify.js). The tests is about 5 times larger than the implementation itself. I think this is a big task. It's [MIT-licensed though](https://github.com/simov/slugify/blob/master/LICENSE) so if, even after pinning the version, we think it's too much of a risk, we could warehouse it (i.e. copy-paste the code into this codebase).

That said, since I mentioned pinning, I think we need to pin it to `1.6.6` instead of `^1.6.6` to avoid any accidental updates.

Reply:
Considering that the use case here is pretty specific, and I can't think of any other feature that would need slugs created, I think using a local function is better.
Changed in be09ed2e30fbabcfd4f54240a1aac360899a2bdc

Reply:
> There seems to be a lot of [edge cases](https://github.com/simov/slugify/blob/master/test/slugify.js).

yeah, but do we need ALL of this functionality ?

Reply:
> > There seems to be a lot of [edge cases](https://github.com/simov/slugify/blob/master/test/slugify.js).
> 
> yeah, but do we need ALL of this functionality ?

Yeah this use case is pretty specifically to slugify an existing URL into another URL. I don't think many of these edge cases applies.

Reply:
> yeah, but do we need ALL of this functionality ?

Haha, hopefully not! Good point.

Honestly though, I frequently see people get burned out for re-inventing the wheel (i.e. [this PR](https://github.com/Automattic/vip-cli/pull/1237) since we roll our own CLI wrapper) rather than not so I tend to automatically side on not re-inventing the wheel unless we have a good reason not to.